internal/configconverter/discovery.go

```diff
@@ -69,10 +69,28 @@ func (Discovery) Convert(_ context.Context, in *confmap.Conf) error {
 		metricsPipeline["receivers"] = appendUnique(metricsReceivers, discoReceivers)
 	}
 
+	setAutoDiscoveryResourceAttribute(service)
+
 	*in = *confmap.NewFromStringMap(out)
 	return nil
 }
 
+func setAutoDiscoveryResourceAttribute(service map[string]any) {
+	telemetry := map[string]any{}
+	if tel, ok := service["telemetry"]; ok {
+		telemetry = tel.(map[string]any)
+	}
+	service["telemetry"] = telemetry
+
+	resAttrs := map[string]any{}
+	if ra, ok := telemetry["resource"]; ok {
+		resAttrs = ra.(map[string]any)
+	}
+	telemetry["resource"] = resAttrs
+
+	resAttrs["splunk_autodiscovery"] = "true"
```

Comment:
should we make it more inline with otel convention and have it be `splunk.autodiscovery`?

Reply:
It's passed through the prometheus, so it'll be replaced with `splunk_autodiscovery` anyway.